mattermost/mattermost-plugin-mscalendar
# Revert "[GH-14] Add at-rest encryption for OAuth2 token (#143)"

This reverts PR [#143 Add at-rest encryption for OAuth2 token](https://github.com/mattermost/mattermost-plugin-mscalendar/pull/143)

#### Summary

With the OAuth encryption code active, the `expires` field in the stored OAuth token is always being set to 0001-01-01T00:00:00Z, when it should be set to time of `connect time + one hour`, something like `2020-06-01T00:14:55.338026-04:00`. This causes Go's OAuth library to think the token is always valid and does not need a refresh.

Since the OAuth tokens expire after one hour, the token stops working after an hour. And since the OAuth library thinks the token is valid, it will not refresh. So no operations work after an hour from connecting.

We've also determined that encrypting the OAuth token brings little to no value because of how the encryption key is stored. It can also make debugging a little more difficult due to not having direct access to the relevant data in the KV store.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-mscalendar/issues/155

#### Test Steps

- Deploy two different builds on two different servers, one on master, one with this PR.
- Connect a user with both builds. After one hour, the master build should show the failures, but the reverted PR build will not have any issues.
